### DISCUSSION TITLE
A new  write_solution_style

### DISCUSSION THREAD

**Discussion by pjaborges** [ERGO-Code/HiGHS#1054] [Ideas]
Is it possible to have a much simpler solution style where only variables with value different from zero are printed?

I am not sure if this is already the case for MIPs since I didn't try yet a MIP on HiGHS.

But for the LP it can be quite handy if no sensitivity analysis or checks for multiple solutions will be performed.

For my specific usage, the problems I handle have as solution for the vast majority of the variables the value zero.
Such a format will be great since files will be way smaller and the parsing them back to my system will be faster.

Thanks!

**Comment by jajhall** [comment]
Anyone who solves problems with vast numbers of variables bounded below by zero will have this issue. If you're solving an LP, the solution file will also contain the dual values, and these will generally be nonzero for the variables at zero, so their number means that the file size isn't reduced significantly. Ditto the basis information, although this could be reduced by only recording the basic variables and nonbasic constraints.

**Reply by pjaborges** [reply]
Can't agree more with the explanations. ;)

> Ditto the basis information, although this could be reduced by only recording the basic variables and nonbasic constraints.

It will be fantastic if this is made available somehow in a future release.

**Reply by jajhall** [reply]
I've largely implemented this if you only need the solution values? Is this the case?

**Comment by jajhall** [comment]
This discussion is answered by the addition of `kSolutionStyleSparse` (value 4) to the `enum SolutionStyle` as a value for option `write_solution_style`, or argument to `Highs::writeSolution`
